app/mappers/status_mapper.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class StatusMapper
+  def self.map(status)
+    if status.nil?
+      status.to_s
+    elsif status.casecmp('in press').zero? || status.casecmp('accepted/in press').zero?
+      'In Press'
+    elsif status.casecmp('published').zero?
+      'Published'
+    else
+      status.to_s
+    end
+  end
```

Comment:
This might be simpler:

``` ruby
    status ||= ''                                                                                                          
                                                                                                                           
    case                                                                                                                   
    when status.match?(/published/i)                                                                                       
      'Published'                                                                                                          
    when status.match?(/in press/i)                                                                                        
      'In Press'                                                                                                           
    else                                                                                                                   
      status.to_s                                                                                                          
    end 
```

You also might not want to call it `map` because that implies an array operation.

Reply:
Another way to do this:

``` ruby
def self.map(status)                                                                                                     
    case (status || '').to_s                                                                                               
    when /published/i                                                                                                      
      'Published'                                                                                                          
    when /in press/i                                                                                                       
      'In Press'                                                                                                           
    else                                                                                                                   
      status.to_s                                                                                                          
    end                                                                                                                    
  end      
```